### ISSUE TITLE
Epic: Storage volume scheduling and movement PoC

### ISSUE THREAD

**Issue by guymguym** [kcp-dev/contrib-tmc#112]
### TL;DR

First attempt, as a Proof of Concept, of managing PVCs and PVs through KCP for scheduling, placement and movement.
Out of scope:
- RWO volumes - volume replication + workload quiesce.
- Placement with multiple storage classes.
- Placement with multiple locations.

### Progress tracking <a href="#progress-tracking" id="progress-tracking">#</a>

EPIC detailed issues and overall progress can be tracked with the following [project view](https://github.com/orgs/kcp-dev/projects/1/views/40)

### Detailed description

The goal and principle is explain in the [following design sketch](https://docs.google.com/drawings/d/1tF1vOurtTW_PzJZbeu5dJrXogX12X51g5Dpz0P2qEFM)

The complete design of the underlying feature relies on Syncer transformations and and Invert Syncing virtual workspace (see [this drawing](https://docs.google.com/drawings/d/1ODyRwO8RbvzDxO90jpKjYvo1KC4m-LsQqgkRtz1cQJw)) 

However until these feature are merged and available, this EPIC is a POC based on [this provisional design](https://docs.google.com/drawings/d/1u_UgLtStPHRZweRnJZr5DvmsXFVSLvsfrp_rn6FKfik/edit)  

Here is a first sketch of the involved control flow: https://docs.google.com/document/d/1lxrF_9Kzxh3IWO66Ec0WNCR-mD0U33gACDeKJWqIO0c

# Demo Objective

- [ ] Admin sets up the control plane with several workload clusters in the same location, connected to the same shared network storage (e.g NFS server).
- [ ] User deploys an application that uses dynamic storage provisioning (e.g [DokuWiki helm chart](https://github.com/bitnami/charts/tree/master/bitnami/dokuwiki)).
- [ ] The application placement is relocated between clusters freely upon cluster failure or on demand with the storage volumes.
- [ ] Deleting the application and storage will de-provision both workload and storage.

# Demo Steps

1. Admin setup shared NFS storage.
2. Admin setup workload clusters with NFS CSI provisioner.
3. Users deploying an application with PVC will sync to a workload cluster for dynamic provisioning.
4. Syncer will watch provisioned volume and sync up PV information to kcp.
5. Cluster failure triggers an update to workload placement.
6. Another workload cluster will sync the PVC+PV bound to each other (static provisioning mode) so that the migrated workload can start.
7. Deleting the namespace will result de-provisioning of the storage as well.

# Action Items 

- [x] Scope the current demo as necessary to fit in prototype boundaries
- [ ] Contribute to final demo script and recording

# Stories

- [x] kcp-dev/kcp#1979
- [x] kcp-dev/kcp#1980
- [ ] kcp-dev/contrib-tmc#98
- [ ] kcp-dev/contrib-tmc#97
- [ ] kcp-dev/contrib-tmc#96
- [x] kcp-dev/kcp#1890 
- [ ] Storage transformations functionality
- [ ] Test case - dynamic provisioning
- [ ] Test case - placement change
- [ ] Test case - volume deletion
